File: mili-api/src/main/java/org/leavesmc/leaves/event/bot/BotLoadEvent.java
/*
 * This file is part of Leaves (https://github.com/LeavesMC/Leaves)
 *
 * Leaves is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Leaves is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Leaves. If not, see <https://www.gnu.org/licenses/>.
 */

package org.leavesmc.leaves.event.bot;

import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

/**
 * Call when a fakeplayer loaded
 */
public class BotLoadEvent extends Event implements Cancellable {

    private static final HandlerList handlers = new HandlerList();

    private final String bot;
    private final UUID botUUID;
    private boolean cancel = false;

    public BotLoadEvent(@NotNull final String who, @NotNull final UUID uuid) {
        this.bot = who;
        this.botUUID = uuid;
    }

    @NotNull
    public static HandlerList getHandlerList() {
        return handlers;
    }

    /**
     * Gets the fakeplayer name
     *
     * @return fakeplayer name
     */
    public String getBot() {
        return bot;
    }

    public UUID getBotUUID() {
        return botUUID;
    }

    @Override
    public boolean isCancelled() {
        return cancel;
    }

    @Override
    public void setCancelled(boolean cancel) {
        this.cancel = cancel;
    }

    @Override
    @NotNull
    public HandlerList getHandlers() {
        return handlers;
    }
}

File: mili-api/src/main/kotlin/org/leavesmc/leaves/event/bot/BotLoadEvent.kt
@file:JvmName("BotLoadEventKt")
package org.leavesmc.leaves.event.bot

import org.bukkit.event.Cancellable
import org.bukkit.event.HandlerList
import org.leavesmc.leaves.event.BukkitEvent
import java.util.UUID

class BotLoadEvent(val botName: String) : BukkitEvent(), Cancellable {
    var uuid: UUID? = null
        private set

    constructor(botName: String, uuid: UUID) : this(botName) {
        this.uuid = uuid
    }

    private var _cancelled = false
    override fun isCancelled() = _cancelled
    override fun setCancelled(cancel: Boolean) { _cancelled = cancel }

    companion object {
        @JvmStatic val HANDLERS = HandlerList()
        @JvmStatic fun getHandlerList() = HANDLERS
    }
    override fun getHandlers() = HANDLERS
}
